pages/api/spaces.ts

```diff
@@ -19,14 +29,49 @@ const fetchSpaces = async () => {
   return response.data.data;
 };
 
+const createSpace = async () => {
+  let response;
+
+  try {
+    response = await muxClient.post(`/video/v1/spaces`, {
+      passthrough: TEMPORARY_SPACE_PASSTHROUGH,
+    });
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      throw new Error(`Error: ${error.response?.status}`);
+    }
+
+    throw new Error("Unknown error");
+  }
+
+  return response.data.data;
+};
+
 export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse
 ) {
-  if (req.method === "GET") {
-    const spaces = await fetchSpaces();
-    res.status(StatusCodes.OK).json(spaces);
+  if (req.method === "POST") {
+    const activeSpaceLimit = process.env.ACTIVE_SPACE_LIMIT;
+    if (activeSpaceLimit) {
+      const limit = parseInt(activeSpaceLimit, 10);
+      const spaces = await fetchSpaces();
+      const activeSpaces = spaces.filter(({ status }) => status === "active");
+      if (activeSpaces.length >= limit) {
+        res.status(StatusCodes.INSUFFICIENT_SPACE_ON_RESOURCE).end();
```

Comment:
This should be returning here as we don't want to enter the `try-catch` block.

Reply:
Oh, good point. I assumed `.end()` returned, but it just closes it for writing and throws an error if something gets written afterwards.